Code/Editor/Controls/FolderTreeCtrl.cpp

```diff
@@ -16,87 +15,76 @@
-
+    , m_folderIcon(":/TreeView/folder-icon.svg")
+    , m_fileIcon(":/TreeView/default-icon.svg")
+    , m_model(new QStandardItemModel(this))
+    , m_proxyModel(new QSortFilterProxyModel(this))
 {
-}
+    m_proxyModel->setSourceModel(m_model);
+    m_proxyModel->setRecursiveFilteringEnabled(true);
+    setModel(m_proxyModel);
 
+    QObject::connect(this, &QTreeView::doubleClicked, this, &CFolderTreeCtrl::OnIndexDoubleClicked);
+}
 
-void CFolderTreeCtrl::init(const QStringList& folders, const QString& fileNameSpec, const QString& rootName, bool bDisableMonitor /*= false*/, bool bFlatTree /*= true*/)
+void CFolderTreeCtrl::Configure(
+    const AZStd::vector<QString>& folders, const QString& fileNameSpec, const QString& rootName, bool bDisableMonitor, bool bFlatTree)
 {
-    m_model = new QStandardItemModel(this);
-    m_proxyModel = new QSortFilterProxyModel(this);
-    m_proxyModel->setRecursiveFilteringEnabled(true);
-
-    m_proxyModel->setSourceModel(m_model);
-    setModel(m_proxyModel);
+    m_folders.clear();
+    AZStd::ranges::copy_if(
+        folders,
+        AZStd::back_inserter(m_folders),
+        [](const auto& path)
+        {
+            if (CFileUtil::PathExists(path) || Path::IsFolder(path.toLocal8Bit().constData()))
+            {
+                return true;
+            }
+            return false;
+        });
+    AZStd::transform(m_folders.begin(), m_folders.end(), m_folders.begin(), 
+        [](auto& item) {
+            return Path::RemoveBackslash(Path::ToUnixPath(item));
+        });
 
-    m_folders = folders;
     m_fileNameSpec = fileNameSpec;
     m_rootName = rootName;
     m_bDisableMonitor = bDisableMonitor;
     m_bFlatStyle = bFlatTree;
-    m_fileIcon = QIcon(":/TreeView/default-icon.svg");
-    m_folderIcon = QIcon(":/TreeView/folder-icon.svg");
 
-    for (auto item = m_folders.begin(), end = m_folders.end(); item != end; ++item)
+    if (m_rootTreeItem)
     {
-        (*item) = Path::RemoveBackslash(Path::ToUnixPath((*item)));
-
-        if (CFileUtil::PathExists(*item))
-        {
-            m_foldersSegments.insert(std::make_pair((*item), Path::SplitIntoSegments((*item)).size()));
-        }
-        else if (Path::IsFolder((*item).toLocal8Bit().constData()))
-        {
-            m_foldersSegments.insert(std::make_pair((*item), Path::SplitIntoSegments((*item)).size()));
-        }
-        else
-        {
-            (*item).clear();
-        }
+        delete m_rootTreeItem;
+        m_rootTreeItem = nullptr;
     }
 
+    m_rootTreeItem = new CTreeItem(*this, nullptr, QString(), m_rootName, IconType::FolderIcon);
+    m_model->invisibleRootItem()->appendRow(m_rootTreeItem);
+
     setHeaderHidden(true);
 
-    QObject::connect(this, &QTreeView::doubleClicked, this, &CFolderTreeCtrl::OnIndexDoubleClicked);
+    for (auto& item: m_folders)
+    {
+        if (!item.isEmpty())
+        {
+            LoadTreeRec(item);
+        }
+    }
 
-    InitTree();
+    if (!m_bDisableMonitor)
```

Comment:
Suggestion: rename the variable to "m_bEnableMonitor" and flip the logic. Every time it's used it's causing a double-negative ("if not disabled"), which is a bit harder to understand than a single-positive ("if enabled"). :)